sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactEditAuthorization.java

```diff
@@ -0,0 +1,64 @@
+package de.symeda.sormas.backend.contact;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.ejb.EJB;
+import javax.ejb.LocalBean;
+import javax.ejb.Stateless;
+
+import de.symeda.sormas.api.user.UserRole;
+import de.symeda.sormas.backend.caze.Case;
+import de.symeda.sormas.backend.caze.CaseEditAuthorization;
+import de.symeda.sormas.backend.caze.CaseService;
+import de.symeda.sormas.backend.region.District;
+import de.symeda.sormas.backend.region.Region;
+import de.symeda.sormas.backend.user.User;
+import de.symeda.sormas.backend.user.UserService;
+
+
+@Stateless (name = "ContactEditAuthorization")
+@LocalBean
+public class ContactEditAuthorization {
+	
+	@EJB
+	private UserService userService;
+	@EJB
+	private CaseService caseService;
+	@EJB
+	private CaseEditAuthorization caseEditAuthorization;
+	@EJB
+	private ContactService contactService;
+	
+    public boolean isContactEditAllowed(String contactUuid){
+
+    	User user = userService.getCurrentUser();
+    	Contact contact = contactService.getByUuid(contactUuid);
+
+        if (user.getUuid().equals(contact.getReportingUser().getUuid())){
+            return true;
+        }
+
+        if (contact.getUuid() != null) {       
+            Case caseofContact = caseService.getByUuid(contact.getUuid());
+            if (caseofContact != null && caseEditAuthorization.caseEditAllowedCheck(contact.getUuid())) {
```

Comment:
You need to use the case here, not the contact.
In addition it would by better to passt the existing contact instance into the check method.
My suggestion is to make caseEditAllowedCheck use an entity as param instead of the uuid. For convenience you can add wrapper method that takes a uuid as param and then calls the other method.

Reply:
I've change the condition to check the case of the contact instead of the contact 
Changed the parameters of both check methods to Cases or Contact entities